packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -220,7 +225,12 @@ export class TabBarRenderer extends TabBar.Renderer {
             const paddingBottom = data.paddingBottom || 0;
             height = `${labelHeight + iconHeight + paddingTop + paddingBottom}px`;
         }
-        return { zIndex, height };
+        if (data.tabWidth) {
+            width = `${data.tabWidth}px`;
+        } else {
+            width = '';
```

Comment:
I remember that the tabs would not get resized when switching from "original" to "dynamic" tabs. My explanation would be that the virtual dom support does a merge of the dom rather than a replace, thus keeping the "width" attribute.

Reply:
I double-checked, we need it.